python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -688,8 +703,191 @@ def asset_check_spec(self) -> AssetCheckSpec:
+                    f" op_execution_context.{attr}."
+                ),
+                breaking_version="1.7",
+                stacklevel=1,
+            )
+
+        if attr in DEPRECATED_IO_MANAGER_CENTRIC_CONTEXT_METHODS:
+            alt = DEPRECATED_IO_MANAGER_CENTRIC_CONTEXT_METHODS[attr]
+
+            # warnings.warn(
+            deprecation_warning(
+                subject=f"AssetExecutionContext.{attr}",
+                additional_warn_text=(
+                    f"You have called method {attr} on AssetExecutionContext that is oriented"
+                    f" around I/O managers. If you not using I/O managers we suggest you {alt}. If"
+                    " you are using I/O managers the method still exists at"
+                    f" op_execution_context.{attr}."
+                ),
+                breaking_version="1.7",
+                stacklevel=1,
+            )
+
+        if attr in ALTERNATE_AVAILABLE_METHODS:
+            deprecation_warning(
+                subject=f"AssetExecutionContext.{attr}",
+                additional_warn_text=f"Instead {ALTERNATE_AVAILABLE_METHODS[attr]}.",
+                breaking_version="1.7",
+                stacklevel=1,
+            )
+
+        return getattr(self._op_execution_context, attr)
+
+    # include all supported methods below
+
+    @public
+    @property
+    def op_execution_context(self) -> OpExecutionContext:
+        return self._op_execution_context
+
+    @public
+    @property
+    def run_id(self) -> str:
+        return self._op_execution_context.run_id
+
+    @public
+    @property
+    def dagster_run(self) -> DagsterRun:
+        """PipelineRun: The current pipeline run."""
+        return self._step_execution_context.dagster_run
+
+    @public
+    @property
+    def asset_key(self) -> AssetKey:
+        return self._op_execution_context.asset_key
+
+    @public
+    @property
+    def pdb(self) -> ForkedPdb:
+        return self._op_execution_context.pdb
+
+    @public
+    @property
+    def log(self) -> DagsterLogManager:
+        """DagsterLogManager: The log manager available in the execution context."""
+        return self._op_execution_context.log
+
+    @public
+    # renaming to avoid ambiguity in single run and multi-partition case
+    @property
+    def is_partitioned_execution(self) -> bool:
+        return self._op_execution_context.has_partition_key
+
+    @public
+    def log_event(self, event: UserEvent) -> None:
+        return self._op_execution_context.log_event(event)
+
+    @public
+    @property
+    def assets_def(self) -> AssetsDefinition:
+        return self._op_execution_context.assets_def
+
+    @public
+    # TODO confirm semantics in the presense of asset subsetting
+    # seems like there should be both "asset_keys" and "selected_asset_keys"
+    @property
+    def selected_asset_keys(self) -> AbstractSet[AssetKey]:
+        return self._op_execution_context.selected_asset_keys
```

Comment:
we also likely will be adding selected_asset_checks here. This refactor might be an opportunity to condense them to some selection object